### PR TITLE
#137156965 Cancel button feature

### PR DESCRIPTION
**Pull request by sayidu**
### What does this PR do?
This PR integrates a cancel option for admin and guest users.

### Description of Task to be completed?
Users should have the option to cancel orders, if the order has not been completed and shipped. 

### How should this be manually tested?
By running the application, creating orders and cancelling them before completion. 
This can be tested from the perspective of a guest and admin user.

### What are the relevant pivotal tracker stories?
#137156965 Add "cancel order" option.

### CODE REVIEW

**Review by tomily1 — COMMENTED** [review]
LGTM

File: imports/plugins/core/checkout/client/templates/checkout/completed/completed.js [hunk 1]
```diff
@@ -32,6 +32,12 @@ Template.cartCompleted.helpers({
     }
     return this.workflow.status;
   },
+  orderCancel: function () {
+    if (this.workflow.status === "coreOrderWorkflow/canceled") {
```
Comment (adebayo-maborukoje) [inline imports/plugins/core/checkout/client/templates/checkout/completed/completed.js:33]: you can just return the boolean from this expression.
```(this.workflow.status === "coreOrderWorkflow/canceled")``` 
There is no need for `If (true) return true;`

File: imports/plugins/core/orders/client/templates/list/items.js [hunk 2]
```diff
@@ -7,6 +7,14 @@ import { NumericInput } from "/imports/plugins/core/ui/client/components";
  *
  */
 Template.ordersListItems.helpers({
+  orderCancel: function () {
```
Comment (adebayo-maborukoje) [inline imports/plugins/core/orders/client/templates/list/items.js]: can we use es6 feature here too?

File: imports/plugins/core/orders/client/templates/list/ordersList.js [hunk 3]
```diff
@@ -32,5 +32,10 @@ Template.dashboardOrdersList.helpers({
   shopName() {
     const shop = Shops.findOne(this.shopId);
     return shop !== null ? shop.name : void 0;
+  },
+  orderCancel() {
+    if (this.workflow.status === "coreOrderWorkflow/canceled") {
```
Comment (adebayo-maborukoje) [inline imports/plugins/core/orders/client/templates/list/ordersList.js]: just return the boolean here as mentioned previously.

File: imports/plugins/core/orders/client/templates/orders.js [hunk 4]
```diff
@@ -191,7 +194,7 @@ Template.ordersListItem.events({
     event.preventDefault();
     const instance = Template.instance();
     const isActionViewOpen = Reaction.isActionViewOpen();
-    const { order } = instance.data;
+    const {order } = instance.data;
```
Comment (adebayo-maborukoje) [inline imports/plugins/core/orders/client/templates/orders.js]: you will need the space between the { and Order here

File: imports/plugins/core/orders/client/templates/workflow/shippingInvoice.html [hunk 5]
```diff
@@ -94,6 +94,9 @@
     <hr>
 
     <div class="flex">
+      {{#if orderCancelled }}
+          <button class="btn btn-danger  flex-item-fill" type="button">Cancelled Order</button>
```
Comment (adebayo-maborukoje) [inline imports/plugins/core/orders/client/templates/workflow/shippingInvoice.html]: How many indentations are we using on the view? Also, there seems to be much space between `danger` and `Flex`

File: imports/plugins/core/orders/client/templates/workflow/shippingTracking.js [hunk 6]
```diff
@@ -109,6 +109,14 @@ Template.coreOrderShippingTracking.helpers({
     return completedItems;
   },
 
+  orderCancelled() {
```
Comment (adebayo-maborukoje) [inline imports/plugins/core/orders/client/templates/workflow/shippingTracking.js:106]: Let's improve this to a simpler method. by returning the boolean only
 e.g `const o = (check) => {return check !== 'ehehe'}`

File: package.json [hunk 7]
```diff
@@ -39,7 +39,7 @@
     "fibers": "^1.0.14",
     "font-awesome": "^4.6.3",
     "griddle-react": "^0.6.1",
-    "i18next": "^3.4.3",
+    "i18next": "^3.5.2",
```
Comment (adebayo-maborukoje) [inline package.json]: any reason for this upgrade?

File: imports/plugins/core/orders/client/templates/workflow/shippingTracking.html [hunk 8]
```diff
@@ -1,5 +1,5 @@
  <template name="coreOrderShippingTracking">
-
+{{#if orderCancelled}}
```
Comment (adebayo-maborukoje) [inline imports/plugins/core/orders/client/templates/workflow/shippingTracking.html]: ### Nitpick
This line is not properly indented

**Review by adebayo-maborukoje — APPROVED** [review]

**Pull request merged**